include/boost/di.hpp

```diff
@@ -1717,6 +1722,11 @@ struct pool<aux::type_list<TArgs...>> : TArgs... {
   pool(const aux::type_list<Ts...>&, TPool p) noexcept : pool(static_cast<Ts&&>(p)...) {
     (void)p;
   }
+  template <class T>
+  pool& operator=(T&& other) noexcept {
+    [](...) {}((static_cast<TArgs&>(*this).operator=(static_cast<TArgs&&>(other)), 0)...);
```

Comment:
can we use swallow instead? The lambda trick doesn't gurantee the order and it's slower to compile than the swallow trick.

```cpp
(void)aux::swallow{0, (static_cast<TArgs&>(*this).operator=(static_cast<TArgs&&>(other)), 0)...};
```